pkg/common/cns-lib/volume/manager.go

```diff
@@ -126,14 +127,18 @@ func GetManager(ctx context.Context, vc *cnsvsphere.VirtualCenter) Manager {
 	}
 	log.Infof("Initializing new volume.defaultManager...")
 	managerInstance = &defaultManager{
-		virtualCenter: vc,
+		virtualCenter:              vc,
```

Comment:
I'm assuming the defaultManager implementation will have many if-statements that checks `idempotencyHandlingEnabled` and have some logic based on whether it is true or false. Have you analyzed if its easier to just define a new `VolumeManager` in a new file that works with operationStore by default?  This way you can use the fss to decide where you need the current `defaultManager` or the new `VolumeManager`.

Reply:
Sure, wanted to make sure if it was analyzed. I would be curious to see how the default manager would look like.